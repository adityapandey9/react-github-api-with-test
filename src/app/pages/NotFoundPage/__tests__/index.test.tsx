import * as React from 'react';
import { NotFoundPage } from '..';
import { BrowserRouter, Switch } from 'react-router-dom';
import { HelmetProvider } from 'react-helmet-async';
import renderer from 'react-test-renderer';
import { Link } from 'app/components/Link';
import { Provider } from 'react-redux';
import { Store } from '@reduxjs/toolkit';
import { configureAppStore } from '../../../../store/configureStore';
import { ThemeProvider } from '../../../../styles/theme/ThemeProvider';

const renderPage = (store: Store) =>
  renderer.create(
    <Provider store={store}>
      <ThemeProvider>
        <HelmetProvider>
          <BrowserRouter>
            <Switch>
              <NotFoundPage />
            </Switch>
          </BrowserRouter>
        </HelmetProvider>
      </ThemeProvider>
    </Provider>,
  );

describe('<NotFoundPage />', () => {
  let store: ReturnType<typeof configureAppStore>;
  let component: ReturnType<typeof renderPage>;

  beforeEach(() => {
    store = configureAppStore();
    component = renderPage(store);
  });
  afterEach(() => {
    component.unmount();
  });

  it('should match snapshot', () => {
    component = renderPage(store);
    expect(component).toMatchSnapshot();
  });

  it('should should contain Link', () => {
    component = renderPage(store);
    expect(component.root.findByType(Link)).toBeDefined();
  });
});
