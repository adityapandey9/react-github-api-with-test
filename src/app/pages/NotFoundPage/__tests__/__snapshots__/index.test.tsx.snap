// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should match snapshot 1`] = `
Array [
  .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  font-size: 1.25rem;
  color: rgba(58,52,51,1);
  font-weight: bold;
  margin-right: 1rem;
}

.c8 {
  text-transform: uppercase;
  font-weight: normal;
  margin: 0;
  padding: 0;
  color: rgba(58,52,51,0.7);
  font-size: 0.75rem;
}

.c10 input[type='radio'] {
  margin: 0;
  opacity: 0;
  width: 0;
  height: 0;
  padding: 0;
}

.c10 input[type='radio'] + label {
  margin: 0;
  display: inline-block;
  padding-left: 1.375rem;
  padding-top: 0.0625rem;
  position: relative;
  cursor: pointer;
  font-size: 0.875rem;
  color: rgba(58,52,51,1);
  z-index: 1;
}

.c10 input[type='radio'] + label a {
  color: rgba(58,52,51,1);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10 input[type='radio'] + label::before {
  position: absolute;
  top: 0.25rem;
  left: 0;
  display: inline-block;
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  content: '';
  border: 2px solid rgba(58,52,51,0.12);
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
}

.c10 input[type='radio'] + label::after {
  display: none;
  content: '';
  position: absolute;
  display: inline-block;
  width: 0.375rem;
  height: 0.375rem;
  border-radius: 50%;
  top: 0.5625rem;
  left: 0.3125rem;
  background-color: rgba(255,255,255,1);
}

.c10 input[type='radio'] + label:hover::before {
  border-color: rgba(215,113,88,1);
}

.c10 input[type='radio']:disabled + label {
  opacity: 0.6;
  cursor: auto;
}

.c10 input[type='radio']:disabled + label:hover::before {
  border-color: rgba(58,52,51,0.12);
}

.c10 input[type='radio']:focus + label::before {
  box-shadow: 0 0 0 3px rgba(215,113,88,0.2);
}

.c10 input[type='radio']:checked + label::before {
  background-color: rgba(215,113,88,1);
  border-color: rgba(215,113,88,1);
}

.c10 input[type='radio']:checked + label::after {
  display: inline-block;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 .c7 {
  margin-bottom: 0.625rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 .radio {
  margin-right: 1.5rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 .c7 {
  margin-bottom: 0.625rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 .radio {
  margin-right: 1.5rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: -1rem;
}

.c2 {
  width: 960px;
  margin: 0 auto;
  padding: 0 1.5rem;
  box-sizing: content-box;
}

.c0 {
  box-shadow: 0 1px 0 0 rgba(58,52,51,0.05);
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  top: 0;
  width: 100%;
  background-color: rgba(255,255,255,1);
  z-index: 2;
}

.c0 .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@supports (-webkit-backdrop-filter:blur(10px)) or (backdrop-filter:blur(10px)) {
  .c0 {
    -webkit-backdrop-filter: blur(10px);
    backdrop-filter: blur(10px);
    background-color: rgba(255,255,255,0.75);
  }
}

<header
    className="c0"
  >
    <div
      className="c1 c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          Github Api with React
        </div>
      </div>
      <nav
        className="c5"
      >
        <div
          className="c6"
        >
          <label
            className="c7 c8"
          >
            Select Language
          </label>
          <div
            className="c9"
          >
            <div
              className="c10 radio"
            >
              <input
                checked={false}
                id="en"
                name="language"
                onChange={[Function]}
                type="radio"
                value="en"
              />
              <label
                htmlFor="en"
              >
                English
              </label>
            </div>
            <div
              className="c10 radio"
            >
              <input
                checked={false}
                id="tr"
                name="language"
                onChange={[Function]}
                type="radio"
                value="de"
              />
              <label
                htmlFor="tr"
              >
                Deutsch
              </label>
            </div>
          </div>
        </div>
        <div
          className="c11"
        >
          <label
            className="c7 c8"
          >
            Select Theme
          </label>
          <div
            className="c12"
          >
            <div
              className="c10 radio"
            >
              <input
                checked={true}
                id="system"
                name="theme"
                onChange={[Function]}
                type="radio"
                value="system"
              />
              <label
                htmlFor="system"
              >
                System theme
              </label>
            </div>
            <div
              className="c10 radio"
            >
              <input
                checked={false}
                id="light"
                name="theme"
                onChange={[Function]}
                type="radio"
                value="light"
              />
              <label
                htmlFor="light"
              >
                Light
              </label>
            </div>
            <div
              className="c10 radio"
            >
              <input
                checked={false}
                id="dark"
                name="theme"
                onChange={[Function]}
                type="radio"
                value="dark"
              />
              <label
                htmlFor="dark"
              >
                Dark
              </label>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </header>,
  .c2 {
  font-size: 1rem;
  line-height: 1.5;
  color: rgba(58,52,51,0.7);
  margin: 0.625rem 0 1.5rem 0;
}

.c3 {
  color: rgba(215,113,88,1);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c3:active {
  opacity: 0.4;
}

.c0 {
  height: calc(100vh - 4rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 320px;
}

.c1 {
  margin-top: -8vh;
  font-weight: bold;
  color: rgba(58,52,51,1);
  font-size: 3.375rem;
}

.c1 span {
  font-size: 3.125rem;
}

@supports (-webkit-backdrop-filter:blur(10px)) or (backdrop-filter:blur(10px)) {

}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      4
      <span
        aria-label="Crying Face"
        role="img"
      >
        ðŸ˜¢
      </span>
      4
    </div>
    <p
      className="c2"
    >
      Page not found.
    </p>
    <a
      className="c3"
      href="/"
      onClick={[Function]}
    >
      Return to Home Page
    </a>
  </div>,
]
`;
